bcda/cli.go

```diff
@@ -117,34 +148,41 @@ func importCCLF9(fileMetadata cclfFileMetadata) error {
 
 func getCCLFFileMetadata(filePath string) (cclfFileMetadata, error) {
 	var metadata cclfFileMetadata
-	// CCLF8/9 filename convention for SSP: P.A****.ACO.ZC*Y**.Dyymmdd.Thhmmsst
+	// CCLF filename convention for SSP: P.A****.ACO.ZC*Y**.Dyymmdd.Thhmmsst
 	// Prefix: T = test, P = prod; A**** = ACO ID; ZC* = CCLF file number; Y** = performance year
-	filenameRegexp := regexp.MustCompile(`(T|P)\.(A\d{4})\.ACO\.ZC(0|8|9)Y\d{2}\.(D\d{6}\.T\d{6})\d`)
+	filenameRegexp := regexp.MustCompile(`(T|P)\.(A\d{4})\.ACO\.ZC(0|8|9)Y(\d{2})\.(D\d{6}\.T\d{6})\d`)
 	filenameMatches := filenameRegexp.FindStringSubmatch(filePath)
 	if len(filenameMatches) < 5 {
 		return metadata, errors.New("invalid filename")
 	}
 
-	filenameDate := filenameMatches[4]
-	t, err := time.Parse("D060102.T150405", filenameDate)
-	if err != nil || t.IsZero() {
-		return metadata, fmt.Errorf("failed to parse date '%s' from filename", filenameDate)
+	cclfNum, err := strconv.Atoi(filenameMatches[3])
+	if err != nil {
+		return metadata, err
 	}
 
-	cclfNum, err := strconv.Atoi(filenameMatches[3])
+	perfYear, err := strconv.Atoi(filenameMatches[4])
```

Comment:
does this return just the year or the entire date string D190417?

Reply:
It grabs the two digits from the `Y**` part of the filename, so for 2019 files it would get just `19`.